tools/PTE/pte-snapshot.js

```diff
@@ -0,0 +1,191 @@
+
+// create client and channel objects
+let client = new hfc();
+let channel = client.newChannel(snapshot.channelID);
+
+// start queryinfo
+queryBlockInfoHandler(channel, client, org, cpf, queryFreq);
+
+/**
+ * query block info
+ *
+ * @param {object} channel The channel object
+ * @param {object} client The client object
+ * @param {string} org The target organization name
+ * @param {string} cpf The connection profile contains the target org
+ * @returns the block height if the block height is less than the target height
+ *          otherwise exit the process 
+ */
+async function queryBlockInfo(channel, client, org, cpf) {
+    const blockchainInfo = await channel.queryInfo();
+    logger.info('[queryBlockInfo] block height= %d', parseInt(blockchainInfo.height));
+
+    return blockchainInfo.height;
+}
+
+/**
+ * query block info callback
+ *
+ * @param {object} channel The channel object
+ * @param {object} client The client object
+ * @param {string} org The target organization name
+ * @param {string} cpf The connection profile contains the target org
+ * @param {int} freq The query info frequency in ms
+ * @returns {Promise<void>}
+ */
+async function queryBlockInfoCB(channel, client, org, cpf, freq) {
+    let blkheight = await queryBlockInfo(channel, client, org, cpf);
+    if ( parseInt(blkheight) > parseInt(snapshot.height) ) {
+        logger.info('[queryBlockInfoCB] block height: target= %d, current= %d', snapshot.height, blkheight);
+        process.exit();
+    }
+
+    // schedule next query
+    setTimeout(function() {
+        queryBlockInfoCB(channel, client, org, cpf, freq);
+    }, freq);
+}
+
+/**
+ * query block info Handler
+ *
+ * @param {object} channel The channel object
+ * @param {object} client The client object
+ * @param {string} org The target organization name
+ * @param {string} cpf The connection profile contains the target org
+ * @param {int} freq The query info frequency in ms
+ * @returns {Promise<void>}
+ */
+async function queryBlockInfoHandler(channel, client, org, cpf, freq) {
+
+    // get client key if clientauth
+    if (TLS == testUtil.TLSCLIENTAUTH) {
+        await testUtil.tlsEnroll(client, org, cpf);
+        logger.debug('[queryBlockInfoHandler] got user private key: org= %s', org);
+    }
+
+
+    hfc.setConfigSetting('key-value-store', 'fabric-common/lib/impl/FileKeyValueStore.js');
+    let cryptoSuite = hfc.newCryptoSuite();
+    cryptoSuite.setCryptoKeyStore(hfc.newCryptoKeyStore({ path: testUtil.storePathForOrg(Nid, org) }));
+    client.setCryptoSuite(cryptoSuite);
+
+    testUtil.assignChannelOrdererSubmitter(channel, client, org, cpPath, TLS);
+
+    let tgtPeers = [];
+    let tgtOrg = [];
+    tgtOrg.push(org);
+    tgtPeers[tgtOrg]=[];
+    tgtPeers[tgtOrg].push(snapshot.peerName);
+    if ( tgtPeers ) {
+        testUtil.assignChannelPeersSubmitter(cpList, channel, client, tgtPeers, TLS, cpPath, null, null, null, null, null);
+    }
```

Comment:
The fact this object can even be constructed is the epitome of everything I hate about Node.js. Regardless I think this can be simplified to:

```
let tgtPeers = [];
tgtPeers[[org]]=[snapshot.peerName];
testUtil.assignChannelPeersSubmitter(cpList, channel, client, tgtPeers, TLS, cpPath, null, null, null, null, null);
```
There is no need to check if `tgtPeers` is null since we explicitly initialize it.

Reply:
ok